doc/tutorials/image/image.txt

```diff
@@ -0,0 +1,185 @@
+   * @param matrix Matrix to save the image from.
+   * @param info An object of ImageInfo class.
+   * @param fatal If an error should be reported as fatal (default false).
+   * @param transpose If true, transpose the matrix after loading.
+   * @return Boolean value indicating success or failure of load.
+   */
+   template<typename eT>
+   bool Load(const std::vector<std::string>& files,
+             arma::Mat<eT>& matrix,
+             ImageInfo& info,
+             const bool fatal,
+             const bool transpose);
+@endcode
+
+@code
+  data::ImageInfo info;
+  std::vector<std::string>> files{"test_image1.bmp","test_image2.bmp"};
+  data::load(files, matrix, info, false, true);
+@endcode
+
+@section save_api_imagetut Save
+
+Save images expects a matrix of type unsigned char in the form (width * height * channels, n_images).
+Just like load it can be used to save one image or multiple images. Besides image data it also expects the shape of the image as input (width, height, channels).
+
+Saving one image:
+
+@code
+  /**
+   * Save the image file from the given matrix.
+   *
+   * @param filename Name of the image file.
+   * @param matrix Matrix to save the image from.
+   * @param info An object of ImageInfo class.
+   * @param fatal If an error should be reported as fatal (default false).
+   * @param transpose If true, transpose the matrix after loading.
+   * @return Boolean value indicating success or failure of load.
+   */
+   template<typename eT>
+   bool Save(const std::string& filename,
+             arma::Mat<eT>& matrix,
+             ImageInfo& info,
+             const bool fatal,
+             const bool transpose);
+@endcode
+
+@code
+  data::ImageInfo info;
+  info.width = info.height = 25;
+  info.channels = 3;
+  info.quality = 90;
+  data::Save("test_image.bmp", matrix, info, false, true);
+@endcode
+
+If the matrix contains more than one image, only the first one is saved.
+
+Saving multiple images:
+
+@code
+  /**
+   * Save the image file from the given matrix.
+   *
+   * @param files A vector consisting of filenames.
+   * @param matrix Matrix to save the image from.
+   * @param info An object of ImageInfo class.
+   * @param fatal If an error should be reported as fatal (default false).
+   * @param transpose If true, transpose the matrix after loading.
+   * @return Boolean value indicating success or failure of load.
+   */
+   template<typename eT>
+   bool Save(const std::vector<std::string>& files,
+             arma::Mat<eT>& matrix,
+             ImageInfo& info,
+             const bool fatal,
+             const bool transpose);
+@endcode
+
+@code
+  data::ImageInfo info;
+  info.width = info.height = 25;
+  info.channels = 3;
+  info.quality = 90;
+  std::vector<std::string>> files{"test_image1.bmp", "test_image2.bmp"};
+  data::Save(files, matrix, info, false, true);
+@endcode
+
+Multiple images are saved according to the vector of file name specified.

```

Comment:
Do you mean "accordingly to the vector of filenames"?

Reply:
Oh!  yes.